### ISSUE TITLE
diesel errors in yagna logs

### ISSUE THREAD

**Issue by jalas167** [golemfactory/yagna#3022]
since ca beginning of this year (2024) I have observed diesel errors.
i.e. after SDK requests to create activity, SDK receives `Service error: DAO error: Diesel error: Cannot perform this operation while a transaction is open.`

As a consequence, providers runs activity, while the requestor assumed it was a failure and rejects payments.

```
[00:15:31.007] [31mERROR[39m: [36mUnable to create activity: Service error: DAO error: Diesel error: Cannot perform this operation while a transaction is open[39m
[00:15:31.008] [33mWARN[39m: [36mStarting task failed due to this issue: Unable to create activity: Service error: DAO error: Diesel error: Cannot perform this operation while a transaction is open[39m
[00:15:31.010] [33mWARN[39m: [36mTask 72 execution failed. Trying to redo the task. Attempt #1. Unable to create activity: Service error: DAO error: Diesel error: Cannot perform this operation while a transaction is open[39m
```

**Comment by grisha87** [comment]
I observed the same behaviour in our production environment.

We're using:
- yagna 0.13.2 downgraded from 0.14.0
- SDK: golem-js 1.0.3 on one application
- yapapi on the other ones

We can't tell where did the issue come, but let me share some insights:

The instnce has been restarted at `2024-01-20T20:34:55.285+0000`

```
2024-01-20 21:34:55.285	
[2024-01-20T20:34:55.285+0000 INFO  yagna] Starting yagna service! Version: 0.13.2 (17c3b70a 2023-11-27 build #356)
```

Then, the first occurrence of the error is at: `2024-01-21T15:20:18.607+0000` - roughly 19 hours difference.

```
2024-01-21 16:20:18.607	
[2024-01-21T15:20:18.607+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
```

The list of errors reported by yagna between these two timestamp is quite short (from newest to oldest):

```
2024-01-21 16:23:31.431	
[2024-01-21T15:23:31.431+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found



2024-01-21 16:22:30.018	
[2024-01-21T15:22:30.018+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:22:27.723	
[2024-01-21T15:22:27.723+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:22:27.670	
[2024-01-21T15:22:27.670+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:22:27.476	
[2024-01-21T15:22:27.476+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:22:22.579	
[2024-01-21T15:22:22.579+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:22:06.314	
[2024-01-21T15:22:06.314+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:21:35.354	
[2024-01-21T15:21:35.354+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:21:32.788	
[2024-01-21T15:21:32.788+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:44.638	
[2024-01-21T15:20:44.638+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:44.568	
[2024-01-21T15:20:44.568+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:44.528	
[2024-01-21T15:20:44.528+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:42.154	
[2024-01-21T15:20:42.154+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:39.533	
[2024-01-21T15:20:39.533+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:19.862	
[2024-01-21T15:20:19.862+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:19.855	
[2024-01-21T15:20:19.855+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:18.779	
[2024-01-21T15:20:18.779+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:18.769	
[2024-01-21T15:20:18.769+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:18.641	
[2024-01-21T15:20:18.641+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 16:20:18.607	
[2024-01-21T15:20:18.607+0000 ERROR ya_activity::error] Activity API server error: DAO error: Diesel error: database is locked
2024-01-21 15:39:57.650	
[2024-01-21T14:39:57.650+0000 ERROR ya_payment::processor] Error sending payment message to provider: Ok(Err(ServiceError("Error while sending payment: Transaction 0x8188c445d40cfe2a010b77d2fdee46be336e63ce9bb51aeb899a325c2bbb16a7 not found on chain")))
2024-01-21 15:17:24.077	
[2024-01-21T14:17:24.077+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
2024-01-21 14:51:42.907	
[2024-01-21T13:51:42.907+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 14:17:31.196	
[2024-01-21T13:17:31.196+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.968	
[2024-01-21T13:15:37.967+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.916	
[2024-01-21T13:15:37.916+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.867	
[2024-01-21T13:15:37.867+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.816	
[2024-01-21T13:15:37.816+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.434	
[2024-01-21T13:15:37.434+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:15:37.381	
[2024-01-21T13:15:37.381+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:14:46.007	
[2024-01-21T13:14:46.007+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 14:13:54.560	
[2024-01-21T13:13:54.560+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 13:53:03.019	
[2024-01-21T12:53:03.018+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 13:21:27.282	
[2024-01-21T12:21:27.282+0000 ERROR ya_payment::processor] Error sending payment message to provider: Ok(Err(ServiceError("Error while sending payment: Transaction 0x4065f25f1fbd1f3f36698c4cf9f07027b3918610d82144a3988d15d51ecf7ccb not found on chain")))
2024-01-21 12:26:40.098	
[2024-01-21T11:26:40.097+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
2024-01-21 11:50:54.457	
[2024-01-21T10:50:54.457+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 09:51:46.435	
[2024-01-21T08:51:46.435+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 09:13:57.021	
[2024-01-21T08:13:57.021+0000 ERROR ya_payment::processor] Error sending payment message to provider: Ok(Err(ServiceError("Error while sending payment: Transaction 0xf4c1152be517a33c9835322f452b693466df28fc8466012aa696155b1dd60632 not found on chain")))
2024-01-21 08:50:36.966	
[2024-01-21T07:50:36.966+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 08:34:24.206	
[2024-01-21T07:34:24.206+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
2024-01-21 07:51:41.327	
[2024-01-21T06:51:41.327+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 07:08:48.477	
[2024-01-21T06:08:48.477+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided



2024-01-21 04:51:05.595	
[2024-01-21T03:51:05.595+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 04:50:56.270	
[2024-01-21T03:50:56.270+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 03:57:41.874	
[2024-01-21T02:57:41.874+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 03:14:05.501	
[2024-01-21T02:14:05.501+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 02:23:30.314	
[2024-01-21T01:23:30.314+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.294	
[2024-01-21T01:21:48.294+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.245	
[2024-01-21T01:21:48.245+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.197	
[2024-01-21T01:21:48.197+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.147	
[2024-01-21T01:21:48.147+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.099	
[2024-01-21T01:21:48.099+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.050	
[2024-01-21T01:21:48.050+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:48.002	
[2024-01-21T01:21:48.002+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:47.951	
[2024-01-21T01:21:47.951+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:47.903	
[2024-01-21T01:21:47.903+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:21:47.853	
[2024-01-21T01:21:47.852+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 02:03:26.365	
[2024-01-21T01:03:26.365+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-21 00:57:30.495	
[2024-01-20T23:57:30.495+0000 ERROR ya_activity::error] Activity API server error: GSB error: GSB failure: Net: error sending message: session closed
2024-01-21 00:50:56.383	
[2024-01-20T23:50:56.383+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-20 23:50:45.143	
[2024-01-20T22:50:45.143+0000 ERROR actix_http::h1::dispatcher] stream error: Request parse error: Invalid Header provided
2024-01-20 21:51:18.069	
[2024-01-20T20:51:18.069+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
2024-01-20 21:51:17.887	
[2024-01-20T20:51:17.887+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
2024-01-20 21:51:17.879	
[2024-01-20T20:51:17.879+0000 ERROR ya_activity::error] Activity API server error: GSB error: Bad request: Bad request: endpoint address not found
```

---

It appears to me that the issue is around the activity creation.

Also, based on polygonscan I can tell that we're super unsuccessfull with scheduling any work on the network. I will restart our yagna instance shortly to see if that's going to help.

**Comment by scx1332** [comment]
Looks like fixed in:
https://github.com/golemfactory/yagna/commit/1d4e18a742d9687a3a1c039a03f93339ccaf49b7
It should be resolved as of 0.15 RC10
